src/features/auth/registaion/RegistraionForm.tsx

```diff
@@ -0,0 +1,134 @@
+import { useState, type FC } from 'react';
+import { Form, Formik, type FormikHelpers } from 'formik';
+import { observer } from 'mobx-react';
+import type { AxiosError } from 'axios';
+import * as Yup from 'yup';
+import { Input } from '@shared/ui/Input';
+import { Button } from '@shared/ui/Button';
+import { Icon } from '@shared/icons/Icon';
+import { useAuth } from '../model/useAuth';
+import type { RegistrationOptions } from './interfaces/registaion-options.interface';
+
+const initialValues: RegistrationOptions = {
+  firstName: '',
+  lastName: '',
+  email: '',
+  password: '',
+  repeatPassword: '',
+};
+
+const RegistrationSchema = Yup.object().shape({
+  firstName: Yup.string().required('Required'),
+  lastName: Yup.string().required('Required'),
+  email: Yup.string().email('Invalid email').required('Required'),
+  password: Yup.string()
+    .min(8, 'Password must be at least 8 characters')
+    .max(20, 'Password must be at most 20 characters')
+    .required('Required'),
+  repeatPassword: Yup.string()
+    .min(8, 'Password must be at least 8 characters')
+    .max(20, 'Password must be at most 20 characters')
+    .oneOf([Yup.ref('password')], 'Passwords must match')
+    .required('Required'),
+});
+
+interface ErrorGeneric {
+  errorCode: string;
+}
+
+export const RegistrationForm: FC = observer(() => {
+  const [isVisiblePassword, setIsVisiblePassword] = useState(false);
+  const { handleSubmitRegistration } = useAuth();
+
+  const handleSubmit = async (
+    values: RegistrationOptions,
+    { setFieldError }: FormikHelpers<RegistrationOptions>
+  ): Promise<void> => {
+    try {
+      await handleSubmitRegistration(values);
+    } catch (error) {
+      const errorObj = error as AxiosError<ErrorGeneric>;
+      if (errorObj.response?.data.errorCode === 'USER_EXISTS') {
```

Comment:
```suggestion
      if (errorResponse?.response?.data?.errorCode === ErrorCodes.ExistedUser) {
```
Should implement enum where value is number